pkg/operator/resource/resourceapply/monitoring.go

```diff
@@ -162,10 +162,14 @@ func ApplyUnstructuredResourceImproved(
 		klog.Infof("%s %q changes: %v", resourceGVR.String(), namespace+"/"+name, JSONPatchNoError(existing, existingCopy))
 	}
 
+	// CRs, unlike certain core objects (e.g., deployments), need to specify the resource version for updates.
+	requiredCopy := required.DeepCopy()
+	requiredCopy.SetResourceVersion(existingCopy.GetResourceVersion())
+
 	// Perform update if resource exists but different from the required (desired) one.
-	actual, err := client.Resource(resourceGVR).Namespace(namespace).Update(ctx, required, metav1.UpdateOptions{})
-	resourcehelper.ReportUpdateEvent(recorder, required, err)
-	cache.UpdateCachedResourceMetadata(required, actual)
+	actual, err := client.Resource(resourceGVR).Namespace(namespace).Update(ctx, requiredCopy, metav1.UpdateOptions{})
```

Comment:
given that `EnsureObjectMeta` and `ensureGenericSpec`modify `existingCopy` and not `requiredCopy` i think we should actually pass `existingCopy` here, am i right ?

Reply:
Right, better to use the `existingCopy` as that also consists of the merged labels, annotations, and owner references, instead of replacing all that with `requiredCopy`'s values. I'd send a patch.

Reply:
ok, please also cover this case in the unit tests (that is make sure the existingCopy is applied).